src/ResourceManager/Storage/Commands.Management.Storage/StorageAccount/StorageAccountBaseCmdlet.cs

```diff
@@ -76,37 +93,97 @@ public string SubscriptionId
+            return returnSkuName;
+        }
+
+        protected static Kind? ParseAccountKind(string accountKind)
+        {
+            Kind returnKind;
+            if (accountKind == null)
+            {
+                return null;
+            }
+            else if (!Enum.TryParse<Kind>(accountKind, true, out returnKind))
+            {
+                throw new ArgumentOutOfRangeException("Kind");
+            }
+            return returnKind;
+        }
+        protected static AccessTier ParseAccessTier(string accessTier)
+        {
+            AccessTier returnAccessTier;
+            if (!Enum.TryParse<AccessTier>(accessTier, true, out returnAccessTier))
             {
-                return AccountType.StandardZRS;
+                throw new ArgumentOutOfRangeException("AccessTier");
             }
-            if (AccountTypeString.StandardGRS.Equals(accountType, StringComparison.OrdinalIgnoreCase))
+            return returnAccessTier;
+        }
+
+        protected static Encryption ParseEncryption(EncryptionSupportServiceEnum? EnableService, EncryptionSupportServiceEnum? DisableService = null)
+        {
+            //DisableService and EnableService should not have overlap
+            if (DisableService != null && EnableService != null)
             {
-                return AccountType.StandardGRS;
+                if ((DisableService & EnableService) != 0)
+                    throw new ArgumentOutOfRangeException("EnableEncryptionService, DisableEncryptionService", String.Format("EnableEncryptionService and DisableEncryptionService should no have overlap Service: {0}", DisableService & EnableService));
             }
-            if (AccountTypeString.StandardRAGRS.Equals(accountType, StringComparison.OrdinalIgnoreCase))
+
+            Encryption accountEncryption = new Encryption();
+            accountEncryption.Services = new EncryptionServices();
+            if (EnableService != null && (EnableService & EncryptionSupportServiceEnum.Blob) == EncryptionSupportServiceEnum.Blob)
             {
-                return AccountType.StandardRAGRS;
+                accountEncryption.Services.Blob = new EncryptionService();
+                accountEncryption.Services.Blob.Enabled = true;
             }
-            if (AccountTypeString.PremiumLRS.Equals(accountType, StringComparison.OrdinalIgnoreCase))
+            if (DisableService != null && (DisableService & EncryptionSupportServiceEnum.Blob) == EncryptionSupportServiceEnum.Blob)
             {
-                return AccountType.PremiumLRS;
+                accountEncryption.Services.Blob = new EncryptionService();
+                accountEncryption.Services.Blob.Enabled = false;
             }
-            throw new ArgumentOutOfRangeException("accountType");
+            return accountEncryption;
+        }
+
+        public static StorageCredentials GenerateStorageCredentials(IStorageManagementClient storageClient, string resourceGroupName, string accountName)
```

Comment:
As there are no one else need to use SRP SDK 5.0, and we are on a tight schedule for the PR, implement the adapter interface for the new storage version might NOT be our priority now.
We can write an adapt in the future when someone else need that.